install_config/aggregate_logging_sizing.adoc

```diff
@@ -0,0 +1,255 @@
+
+
+Rsyslog is configured (see ratelimit.interval, ratelimit.burst) to rate-limit entries read from the journal at 10,000 messages in 300 seconds. A good rule of thumb is to ensure that the rsyslog rate-limits always accommodate to the systemd-journald rate-limits.
+
+[[install-config-aggregate-logging-sizing-guidelines-scaling-up]]
+== Scaling up EFK logging 
+
+If you don't indicate the desired scale at first deployment, the least disruptive way of adjusting your cluster is by re-running the deployer with the updated `*ES_CLUSTER_SIZE*` value and using the `MODE=reinstall` template parameter. Refer to the developers xref:../install_config/aggregate_logging.adoc#aggregate-logging-performing-elasticsearch-maintenance-operations[documentation] for more in depth information about this topic.
+
+----
+$ oc edit configmap logging-deployer
+  [change es-cluster-size value to 5]
+
+$ oc new-app logging-deployer-template --param MODE=reinstall
+----
+
+[[install-config-aggregate-logging-sizing-guidelines-storage]]
+== Storage Considerations 
+An Elasticsearch index is a collection of shards and it's correspondent replica shards. This is how ES implements high availability internally, therefore there is little need of using hardware based mirroring RAID variants. RAID 0 can still be used to increase overall disk performance. +
+Every search request needs to hit a copy of every shard in the index.
+Each ES instance requires it's own individual storage, but an {product-title} deployment can only provide volumes shared by all of its pods, which again means that Elasticsearch shouldn't be implemented with a single node. +
+
+A xref:../install_config/persistent_storage/index.adoc#install-config-persistent-storage-index[persistent volume] should be added to each Elasticsearch deployment configuration so that we have one volume per link:https://www.elastic.co/guide/en/elasticsearch/guide/current/replica-shards.html[replica shard]. On {product-title} this is often achieved through xref:../architecture/additional_concepts/storage.adoc#persistent-volume-claims[Persistent Volume Claims] +
+
+* 1 volume per shard
+
+* 1 volume per replica shard
+  
+The PVCs must be named based on the `es-pvc-prefix` setting. Refer to xref:../install_config/aggregate_logging.adoc#overview[Persistent Elasticsearch Storage] for more details. Below are capacity planning guidelines for {product-title} Aggregated Logging.
+
+*Example scenario*
+
+Assumptions: +
+1) which app: Apache +
+2) bytes per line: 256 +
+3) lines per second load on app: 1 +
+4) raw text data -> json +
+
+
+Logging project:
+
+Baseline (256 characters per minute -> 15KB/min)
+
+----
+
+3 es 
+1 kibana
+1 curator
+1 fluentd
+6 pods: 90000 x 86400 = 7,7 GB/day
+
+
+3 es
+1 kibana
+1 curator
+11 fluentd 
+16 pods: 225000 x 86400 = 24,0 GB/day
+
+3 es
+1 kibana
+1 curator
+20 fluentd 
+25 pods: 225000 x 86400 = 32,4 GB/day
+
+----
+
+A general logging load can be obtained with some simple formulas:
```

Comment:
Calculating total logging throughput and disk space required for your logging environment requires knowledge of your application.  For example, if one of your applications on average logs 10 lines-per-second, each 256 bytes in length, calculate throughput and disk space as follows: